src/core/hle/service/ldn/ldn_types.h

```diff
@@ -148,9 +148,24 @@ struct Ssid {
     u8 length;
     std::array<char, SsidLengthMax + 1> raw;
 
+    Ssid() {
+        length = 0;
+        std::memset(raw.data(), 0, raw.size());
+    }
+
+    Ssid(std::string data) {
+        length = static_cast<u8>(std::min(data.size(), SsidLengthMax));
+        std::memcpy(raw.data(), data.data(), length);
+        raw[length] = 0;
+    }
```

Comment:
```suggestion
    u8 length{};
    std::array<char, SsidLengthMax + 1> raw{};

    Ssid() = default;

    Ssid(std::string_view data) {
        std::strncpy(raw.data(), data.c_str(), SsidLengthMax);
        length = static_cast<u8>(std::min(data.size(), SsidLengthMax));
    }
```
Is storing the length necessary if the string is null terminated?

Reply:
Yes, as this struct has been taken from the Switch LDN service implementation.
Also, std::string_view doesn't support calling c_str() on it.